src/st/no3/3C4EC.c

```diff
@@ -267,8 +267,199 @@ void EntityUnkId50(Entity* self) {
     }
 }
 
+extern u16 D_801817D4[];
+extern u16 D_801817DC[];
+extern u8 D_801817E4[];
+extern u16 D_801817E8[];
+extern u16* D_801817EC[];
 // part of parallax background with pine trees
-INCLUDE_ASM("st/no3/nonmatchings/3C4EC", EntityBackgroundPineTrees);
+void EntityBackgroundPineTrees(Entity* self) {
+    Tilemap* gTilemap = &g_Tilemap;
+    u16 selfUnk68;
+    s32 primIndex;
+    u16* var_s5;
+    u16 var_s4;
+    u16 width;
+    u16 xpos;
+    Primitive* prim_s1;
+    Primitive* prim_s0;
+
+    xpos = self->params; // Temporary reuse of xpos var for params
+    selfUnk68 = D_801817D4[xpos];
+    var_s5 = D_801817EC[xpos];
+    var_s5 += (u16)self->ext.timer.t * 2;
+    var_s4 = D_801817DC[xpos];
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180AD0);
+        self->flags |= FLAG_POS_CAMERA_LOCKED;
+        self->unk68 = selfUnk68;
```

Comment:
It looks like unk68 is a ratio used to determine the amount of x and y scroll https://github.com/Xeeynamo/sotn-decomp/blob/50b81355310b70b4316a3828c5fbd8c2feef485b/src/dra/5087C.c#L1420

Reply:
Agreed, seems right. Naming that should probably be a separate PR, but nice to learn more about how entities work.